### PR TITLE
feat: Add support for event type and custom contexts

### PR DESCRIPTION
**Pull request by HazAT**
A lot of code, but with this, you can send transactions to Sentry.
```swift
        let event = Event(level: .info)
        event.transaction = "test.transaction.1"
        event.type = "transaction"
        let context = Context()
        context.otherContexts = ["trace": [
            "trace_id": NSUUID().uuidString.replacingOccurrences(of: "-", with: "").lowercased(),
            "span_id": NSUUID().uuidString.replacingOccurrences(of: "-", with: "").lowercased().prefix(16),
            "op": "manual",
            "description": "Test operation"
            ]
        ]
        event.context = context
        Client.shared?.send(event: event) { (error) in
            // Optional callback after event has been send
        }
```

### CODE REVIEW

**Comment by dcramer** [comment]
Carry over from Slack:

We may also want to provide timestamp/start_timestamp on the event to complete the high level use case of transactions.

**Comment by HazAT** [comment]
Added possibility to set `startTimestamp` on the event:

```swift
let calendar = Calendar.current
event.startTimestamp = calendar.date(byAdding: .minute, value: -5, to: Date())
```

even though it doesn't show up in the UI anywhere.

**Comment by dcramer** [comment]
:+1: confirming w/ folks if we think this is MVP event that makes transactions usable

**Comment by dcramer** [comment]
@HazAT event.timestamp is set when Event is initialized i assume? is it also possible to set manually?

File: Sources/Sentry/SentryEvent.m [hunk 1]
```diff
@@ -163,6 +163,16 @@ - (void)addSimpleProperties:(NSMutableDictionary *)serializedData {
     [serializedData setValue:self.message forKey:@"message"];
     [serializedData setValue:self.logger forKey:@"logger"];
     [serializedData setValue:self.serverName forKey:@"server_name"];
+    [serializedData setValue:self.type forKey:@"type"];
+    if (nil != self.type && [self.type isEqualToString:@"transaction"]) {
```
Comment (dcramer) [inline Sources/Sentry/SentryEvent.m]: ah nice we're enforcing it

**Comment by dcramer** [comment]
Sounds like the above examples should be fine. Thumbs up from me.

**Pull request merged**